src/cursor.rs

```diff
@@ -66,15 +67,124 @@ impl RawCursor {
 
     fn received_bytes(&self) -> u64 {
         match &self.0 {
-            RawCursorInner::Waiting(_) => 0,
-            RawCursorInner::Loading(state) => state.net_size,
+            RawCursorState::Waiting(_) => 0,
+            RawCursorState::Loading(state) => state.net_size,
         }
     }
 
     fn decoded_bytes(&self) -> u64 {
         match &self.0 {
-            RawCursorInner::Waiting(_) => 0,
-            RawCursorInner::Loading(state) => state.data_size,
+            RawCursorState::Waiting(_) => 0,
+            RawCursorState::Loading(state) => state.data_size,
+        }
+    }
+}
+
+// === BytesCursor ===
+
+/// Unlike [`RowCursor`] which emits rows deserialized as structures from RowBinary,
+/// this cursor emits raw bytes without deserialization.
+/// It can be iterated over using [`futures::StreamExt::next`].
+/// Additionally, if the requested format emits each row on a newline
+/// (e.g. `JSONEachRow`, `CSV`, `TSV`, etc.), the cursor can be converted into a stream of lines
+/// using [`futures::AsyncBufReadExt::lines`]`, which will allow for more convenient processing.
+#[cfg(feature = "futures03")]
+pub struct BytesCursor {
+    raw: RawCursor,
+    buffer: Vec<u8>,
```

Comment:
`buffer` + `filled` + `consumed` is actually `bytes_ext::BytesExt` (already used by other cursors)